ios/brave-ios/Sources/BraveWallet/Crypto/WalletActions/DepositTokenView.swift

```diff
@@ -411,6 +446,21 @@ private struct DepositDetailsView: View {
   }
 }
 
+extension Array where Element == BraveWallet.NetworkInfo {
+  fileprivate func supportedNetworks(
+    coin: BraveWallet.CoinType,
+    keyringId: BraveWallet.KeyringId
+  ) -> [BraveWallet.NetworkInfo] {
+    if coin == .eth || coin == .sol {
+      return self.filter { $0.coin == coin }
+    } else {
+      return self.filter {
+        $0.supportedKeyrings.contains(keyringId.rawValue as NSNumber)
+      }
+    }
+  }
+}
```

Comment:
nit: don't think `coin` needs passed into here 🙂. 

`supportedKeyrings` is assigned using `GetSupportedKeyringsForNetwork`
https://github.com/brave/brave-core/blob/98ba0fd297b46dbdacff70c281e9e571e5f6a287/components/brave_wallet/browser/blockchain_list_parser.cc#L556

Which actually switches on `CoinType` already:
https://github.com/brave/brave-core/blob/98ba0fd297b46dbdacff70c281e9e571e5f6a287/components/brave_wallet/common/common_utils.cc#L208